src/libraries/System.Net.Requests/tests/HttpWebRequestTest.cs

```diff
@@ -2077,19 +2077,115 @@ public async Task SendHttpGetRequest_WithCachePolicyBypassCache_DoNotAddHeaders(
             });
         }
 
+        [Fact]
+        public async Task SendHttpPostRequest_ConnectionShouldStartWithRequestStream()
+        {
+            await LoopbackServer.CreateClientAndServerAsync(
+                async (uri) =>
+                {
+                    HttpWebRequest request = WebRequest.CreateHttp(uri);
+                    request.Method = "POST";
+                    var stream = await request.GetRequestStreamAsync();
+                },
+                async (server) => 
+                {
+                    Exception? exception = await Record.ExceptionAsync(() => server.AcceptConnectionAsync(_ =>
+                    {
+                        return Task.CompletedTask;
+                    }));
+                    Assert.Null(exception);
+                }
+            );
+        }
+
+        [Fact]
+        public async Task SendHttpPostRequest_AllowWriteBufferingFalse_ConnectionShouldStartWithRequestStream()
+        {
+            const string text = "Hello World!!!!\n";
+            await LoopbackServer.CreateClientAndServerAsync(
+                async (uri) =>
+                {
+                    const int size = 1048576 * 2;
+                    HttpWebRequest request = WebRequest.CreateHttp(uri);
+                    request.ContentType = "application/text";
+                    request.ContentLength = size;
+                    request.Method = "POST";
+                    request.AllowWriteStreamBuffering = false;
+                    using var stream = await request.GetRequestStreamAsync();
+                    for (int i = 0; i < size / text.Length; i++)
+                    {
+                        await stream.WriteAsync(new ReadOnlyMemory<byte>(Encoding.UTF8.GetBytes(text)));
+                    }
+                },
+                async (server) =>
+                {
+                    await server.AcceptConnectionAsync(async connection =>
+                    {
+                        var data = await connection.ReadRequestDataAsync();
+                        for (int i = 0; i < data.Body.Length; i += text.Length)
+                        {
+                            Assert.Equal(text, Encoding.UTF8.GetString(data.Body[i..(i + text.Length)]));
+                        }
+                    });
+                }
+            );
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        public async Task SendHttpPost_WriteBufferingEnabled(int delayMs)
```

Comment:
What is this test verifying?
Looks like we're writing & sleeping 16k times, but only checking that the first write went through?